### PR TITLE
completed assessment

### PR DESCRIPTION
**Pull request by mrbrantley**

### CODE REVIEW

File: code-challenges.test.js [hunk 1]
```diff
@@ -11,49 +11,198 @@
...
+// }
+
+
+//Refactored code moved above line 39 to make test pass. Function must be defined in order to pass test. See lines 80-87 for refactored code with psuedo code. 
+const divisibleByThree = (num) => {
+  if (num % 3 === 0) {
+    return `${num} is divisible by three`;
+  }
+  return `${num} is not divisible by three`
+}
```
Comment (WalkerTexas-coder) [inline code-challenges.test.js:40]: good use of the inherent "return" break

File: code-challenges.test.js [hunk 2]
```diff
@@ -11,49 +11,198 @@
...
+
 var num3 = -7
+console.log(divisibleByThree(num3))
 // Expected output: "-7 is not divisible by three"
 
+describe("divisibleByThree", () => {
+  it("returns a string of `-7 is not divisible by three` if a number is divisible by 3 or not based on a number input", () => {
+    expect(divisibleByThree(num3)).toEqual(`-7 is not divisible by three`)
+  })
+})
```
Comment (WalkerTexas-coder) [inline code-challenges.test.js:73]: this is good but I am gonna leave a standard test using what you wrote here for your ref: 

```javascript
describe("divisibleByThree", () => {
  it("returns a string of `15 is divisible by three` if a number is divisible by 3 or not based on a number input", () => {
    expect(divisibleByThree(num1)).toEqual(`15 is divisible by three`)
    expect(divisibleByThree(num2)).toEqual(`0 is divisible by three`)
    expect(divisibleByThree(num3)).toEqual(`-7 is not divisible by three`)
  })
})
```

File: code-challenges.test.js [hunk 3]
```diff
@@ -11,49 +11,198 @@
...
+console.log(capitalPunishment(randomNouns2));
 
 
+describe("capitalPunishment", () => {
+  it("returns a new array with all string elements capitalized", () => {
+    expect(capitalPunishment(randomNouns2)).toContain( "Temperature", "Database", "Chopsticks", "Mango", "Deduction" )
+  })
+})
 
 // b) Create the function that makes the test pass.
```
Comment (WalkerTexas-coder) [inline code-challenges.test.js:123]: Just a heads keep your functions under the B section to give me the illusion (or reality) of writing them after your tests.

File: code-challenges.test.js [hunk 4]
```diff
@@ -11,49 +11,198 @@
...
 
 
 // --------------------2) Create a function that takes in an array of words and returns an array with all the words capitalized.
 
+//Declare a function that iterates over an array
+const capitalPunishment = (array) => {
+  //Use the .map higher order function to apply an operation over every value in the array.
+  return array.map(value => value.charAt(0).toUpperCase() + value.slice(1))
+  //The first index of each string on the array will be the first letter, to change the first letter I used the .charAt(0) and applied the .toUppercase methods to capitalize the first letter. Adding the capitalised letter to the same string I used the .slice method to create a string missing the first index/first letter. .slice(1) on a string will create a string missing the first letter/index. 
+};
```
Comment (WalkerTexas-coder) [inline code-challenges.test.js:97]: Solid work!

File: code-challenges.test.js [hunk 5]
```diff
@@ -11,49 +11,198 @@
...
+//   let vowels = 'aeiouAEIOU'
+//   //My new variable tells the function what all of the vowels are, so now we need to make an operation that tests for these specific characters on the input string. Iteration seems like the best way to do this. 
+//   for (i=0; i<string.length; i++) {
+//     //This will iterate over the string at each letter. Because javascript reads a string similar to an array with each letter of the string having an assigned index, I used the same iteration technique that loops over an array. Starting at the first index/letter [0], iterate over the string up until the last index of the string or the .length of the string, and continue to iterate each time one index higher or index + 1 or index ++. 
+//     for (j=0; j<vowels.length; j++) {
+//       //Because I have two strings that I want to compare to each other, I will use the same iteration method over my new variable of vowels alongside the first iteration. 
+//       //I want to find out the first occurence at which index on input string matches any of the letters on  my vowels variable string. A conditional statement can be used to compare the two strings.
+//       if(string[i]===vowels[j]){
```
Comment (WalkerTexas-coder) [inline code-challenges.test.js:143]: I recognize this!

File: code-challenges.test.js [hunk 6]
```diff
@@ -11,49 +11,198 @@
...
+//   let i = 0
+//   let j = 0
+//   while(i < string.length && j < vowels.length){
+//     if(string[i] === vowels[j]) j++;
+//     i++;
+//   }
+//   return string.indexOf(string[i])
+// };
+
+//My understanding of while statements is still outside of my current skillset. Attempts at using the refactored code continued to return undesireable results that failed the test. So I will stick with my current working version of the code. Refactored without comments on lines 152-161.
```
Comment (WalkerTexas-coder) [inline code-challenges.test.js:208]: Great work pushing yourself

File: coding-concepts.js [hunk 7]
```diff
@@ -8,39 +8,39 @@
...
 const greeter = (name) => {
   `Hello, ${name}!`
 }
-// console.log(greeter("LEARN Student"))
+console.log(greeter("LEARN Student"))
 
-// a) Your answer:
-// b) Verify and explain:
+// a) Your answer: This will log "Hello, LEARN Student". The console takes the function of greeter and logs the argument of name. The function greeter takes the input of name then uses string iterpolation to add the input name string into the phrase "Hello, (input name)!"
+// b) Verify and explain: I was incorrect. It returns undefined because the function needs a return value to log the variable. Without a return value the function doesn't store the data of `Hello, ${name}!` to be called on later.
```
Comment (WalkerTexas-coder) [inline coding-concepts.js:25]: Yes!

File: coding-concepts.js [hunk 8]
```diff
@@ -53,7 +53,10 @@ class Learn {
...
-// console.log(learnStudent)
+console.log(learnStudent)
 
-// a) Your answer:
-// b) Verify and explain:
+// a) Your answer: This will log an object with the following values:
+// ['George', 
+// 'Delta', 
+// 2021]
+// b) Verify and explain: I was 50% correct (or 50% incorrect depending on your outlook). I did not include the keys on the predicted object. The "new" command created a new object with the class of Learn. It took the input of "George" as the name and assigned the value to the "student" key. The "new" command built the object with the "George" input using the "Learn" class blueprint and logged it to the console using the console.log.
```
Comment (WalkerTexas-coder) [inline coding-concepts.js:62]: yea the KEY difference here being learnStudent is an object and not an array.

File: interview-questions.md [hunk 9]
```diff
@@ -8,62 +8,62 @@ Then, research the question to expand on your answer. Even if you feel you have
...
 
 
 
 2. The JavaScript built in method .map() takes predefined parameters. What are they? Which are required and which are optional?
 
-  Your answer:
+  Your answer: The .map higher order function takes in the parameters of value, index, and length. The .map function is intended to work on arrays. Only one parameter is required, but it doesn't matter which one of the three. 
 
-  Researched answer:
+  Researched answer: The .map method takes in parameters of value, index, and length of an array. The value parameter is required and the index and length parameters are optional because the .map method is a higher order function iterates through an array to give an output of a new array and an array cannot exist without defined values. The index and length of the array will be deifined by the input values and number of elements input.
```
Comment (WalkerTexas-coder) [inline interview-questions.md:21]: excellent

File: interview-questions.md [hunk 10]
```diff
@@ -8,62 +8,62 @@ Then, research the question to expand on your answer. Even if you feel you have
...
 
 
 
 4. What is the difference between a function and a method?
 
-  Your answer:
+  Your answer: A function should be universal and can be take various inputs such as defined variables outside of the function. Methods are built into the language of javascript to be used to perform a specific task on a specific or limited data type.
 
-  Researched answer:
+  Researched answer: Functions are blueprint for logic, that are reusable pieces of code that only execute when being called on by its name and that uses an input parameter to execute. Methods are associated with object data types, is invoked using the dot notation or square bracket notation, and operates on the data contained in a class.
```
Comment (WalkerTexas-coder) [inline interview-questions.md:37]: Methods are functions that belong to a obj (generally one assoc with a class) and act upon the object they belong too. (i.e. string methods act upon anything made with the class of string)

File: interview-questions.md [hunk 11]
```diff
@@ -8,62 +8,62 @@ Then, research the question to expand on your answer. Even if you feel you have
...
 
 
 
 5. What is the difference between a class and an object?
 
-  Your answer:
+  Your answer: A class is a blueprint of an object that can be called upon universally throughout the code, similarly how a function operates for logic. An object is a defined "object" data type that stores key: value pairs, stored and structured similar to an array. 
 
-  Researched answer:
+  Researched answer: A class is a blueprint of an object that is universal, reusable, and customizable. Each object created from a defined class is indpenedent from each other object that is created from that same defined class.
```
Comment (WalkerTexas-coder) [inline interview-questions.md:45]: Yes!

**Review by WalkerTexas-coder — APPROVED** [review]
Ross great work! Good use of communication skills via pseudo code and notes. Great work on the logic and implementation of what we went over in office hours. I left some other comments for reference and encouragement below. Keep up the good work your PR is approved!

**Comment by mrbrantley** [comment]
Thanks for the feedback!

**Pull request merged**